lib/Config.js

```diff
@@ -1284,11 +1287,32 @@ class Config extends EventEmitter {
     }
 }
 
+// get the version ID encoding type
+function getVersionIdEncType(verCfg) {
+    if (verCfg === undefined) {
+        throw new Error('Invalid version configuration');
```

Comment:
it looks dangerous to me to throw here, e.g. at https://github.com/scality/cloudserver/pull/3660/files#diff-d9d6c7d27a37666321fe0d79f3d76b7349a03856a0933381651431fb4e5cff62R341 are we sure that there is always a defined versioning configuration?

Reply:
I realize that `decode()` is sometimes called without a versioningConfiguration. Problematic.

Reply:
So if you look carefully I somehow managed to call `getEncType()` only when needed and it does not assert any more.